src/main/java/org/openstreetmap/atlas/checks/validation/points/InvalidMiniRoundaboutCheck.java

```diff
@@ -0,0 +1,113 @@
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flags any nodes tagged as highway=MINI_ROUNDABOUT that do not have very many incoming/outgoing
+ * edges as potentially tagged incorrectly.
+ *
+ * @author nachtm
+ */
+public class InvalidMiniRoundaboutCheck extends BaseCheck<Long>
+{
+
+    private static final long DEFAULT_VALENCE = 6;
+    private static final String MINIMUM_VALENCE_KEY = "minimumValence";
+    private static final String OTHER_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#})"
+            + " has {1, number,#} connecting car-navigable edges. Consider changing this.";
+    private static final String TWO_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#}) "
+            + "has 2 connecting car-navigable edges. Consider changing this to highway=TURNING_LOOP or "
+            + "highway=TURNING_CIRCLE.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(TWO_EDGES_INSTRUCTION,
+            OTHER_EDGES_INSTRUCTION);
+    private static final String DIRECTION_KEY = "direction";
+    private static final List<String> VALID_DIRECTIONS = Arrays.asList("clockwise",
+            "anticlockwise");
+    private final long minimumValence;
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public InvalidMiniRoundaboutCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minimumValence = this.configurationValue(configuration, MINIMUM_VALENCE_KEY,
+                DEFAULT_VALENCE);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Node
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.MINI_ROUNDABOUT);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Node node = (Node) object;
+        final Collection<Edge> carNavigableEdges = getCarNavigableEdges(node);
+        final long valence = carNavigableEdges.size();
+        final Optional<CheckFlag> result;
+
+        if (isTurnaround(node, carNavigableEdges))
+        {
+            result = Optional.of(flagNode(node, carNavigableEdges,
+                    this.getLocalizedInstruction(0, node.getOsmIdentifier())));
+        }
+        else if (!node.containsValue(DIRECTION_KEY, VALID_DIRECTIONS) && valence < minimumValence
+                && valence > 0)
+        {
+            result = Optional.of(flagNode(node, carNavigableEdges,
+                    this.getLocalizedInstruction(1, node.getOsmIdentifier(), valence)));
+        }
+        else
+        {
+            result = Optional.empty();
+        }
+        return result;
+    }
+
+    private CheckFlag flagNode(final Node node, final Collection<Edge> edges,
```

Comment:
All your helper functions should have java docs comments.

Reply:
Thanks for the note! I'm not going to comment on all of these but this round of changes should be addressed in f83ce00.